### PR TITLE
ARMADA-402 Added search and sort for list endpoints

### PR DESCRIPTION
**Pull request by dusktreader**
#### What
Added search and sort capability to list endpoints.

#### Why
This will allow the UI to sort and filter the items in the data tables it shows.

`Task`: https://app.clickup.com/t/18022949/ARMADA-402

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: backend/tests/api/test_config.py [hunk 1]
```diff
@@ -138,6 +138,55 @@ async def test_get_all_configurations__success(
...
+@database.transaction(force_rollback=True)
+async def test_get_all_configurations__with_search(
+    backend_client: AsyncClient,
+    some_configuration_rows,
+    some_configuration_items,
+    insert_objects,
+    inject_security_header,
+):
+    """
+    Test fetching all configuration rows in the db.
```
Comment (julianaklulo) [inline backend/tests/api/test_config.py]:
```suggestion
    Test fetching configuration rows in the db and applying search.
```

File: backend/tests/api/test_config.py [hunk 2]
```diff
@@ -138,6 +138,55 @@ async def test_get_all_configurations__success(
...
+@database.transaction(force_rollback=True)
+async def test_get_all_configurations__with_sort(
+    backend_client: AsyncClient,
+    some_configuration_rows,
+    some_configuration_items,
+    insert_objects,
+    inject_security_header,
+):
+    """
+    Test fetching all configuration rows in the db.
```
Comment (julianaklulo) [inline backend/tests/api/test_config.py]:
```suggestion
    Test fetching configuration rows in the db and sorting.
```

File: backend/tests/test_storage.py [hunk 3]
```diff
@@ -32,3 +45,76 @@ def test_create_all_tables(raw_connection):
...
+        f"""
+        select {columns_str}
+        from config
+        order by
+            config.name desc
+        """
+    )
+
+
+def test_sort_clause__produces_valid_query():
```
Comment (julianaklulo) [inline backend/tests/test_storage.py]: This test has the same name as the test above.

**Review by julianaklulo — CHANGES_REQUESTED** [review]
Looking good, just have a small request change.

File: backend/tests/api/test_config.py (hunk 1 above)
Comment (dusktreader) [reply backend/tests/api/test_config.py]: Fixed in commit 91bdc8f

File: backend/tests/api/test_config.py (hunk 2 above)
Comment (dusktreader) [reply backend/tests/api/test_config.py]: Fixed in commit 91bdc8f

File: backend/tests/test_storage.py (hunk 3 above)
Comment (dusktreader) [reply backend/tests/test_storage.py]: Fixed in commit 91bdc8f

**Review by julianaklulo — APPROVED** [review]

**Pull request merged**